File: Sources/PySwift/Bridging/PYSwiftBridinglayer.swift
//
//  File.swift
//  
//
//  Created by Nevio Hirani on 03.02.24.
//

import Foundation

public class PYSwiftBridinglayer {
    public func executePythonScript(script: String) {
        let process = Process()
        process.launchPath = "/usr/bin/env"
        process.arguments = ["python", "-c", script]
        process.launch()
        process.waitUntilExit()
    }
}

File: Sources/PySwift/lib/include/PYSwiftBridinglayer.c
//
//  PYSwiftBridinglayer.c
//  
//
//  Created by Nevio Hirani on 03.02.24.
//

#include <stdio.h>
#include "PYSwiftBridginglayer.h"
#include <Python/Python.h>

void execute_python_script(const char *script) {
    Py_Initialize();
    PyRun_SimpleString(script);
    Py_Finalize();
}
